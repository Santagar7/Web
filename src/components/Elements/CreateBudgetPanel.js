import axios from "axios";
import React from 'react';
axios.defaults.withCredentials = true;

function CreateBudgetPanel({active,func}){
    const url = 'http://localhost:7000/';
    const CreateBudget = ()=>{
        let name = document.getElementById('name').value;
        let elem = document.getElementById('message');
        axios.post(url+'create_budget',{
            'name':name
        }).then((resp)=>{
            if(resp.data.message==='Success'){
                navigator("/budgets");
            }
            else{
                elem.textContent = resp.data.message;
                elem.style.color = 'crimson';
            }
        })
    }
    return(
        <div className={active?"create_budget":"create_budget not_active"}>
            <div className="center">
                <h1 id="message">Create new budget</h1>
                <form>
                    <input id="name" type="text" required  data-testid='name'/>
                    <button onClick={CreateBudget}>Submit</button>
                </form>
                <button >Cancel</button>
            </div>
        </div>
    )
}

export default CreateBudgetPanel;
