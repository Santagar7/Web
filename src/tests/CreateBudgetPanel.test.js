import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import CreateBudgetPanel from '../components/Elements/CreateBudgetPanel';
import renderer from 'react-test-renderer';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><CreateBudgetPanel/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
test('test login data', () => {
    render(
        <BrowserRouter>
            <CreateBudgetPanel />
        </BrowserRouter>,
    );
    const inputName = screen.getByTestId('name');
    fireEvent.change(inputName, { target: { value: 'steve' } });
    expect(inputName.value).toBe('steve');
});

test('test submit btn', () => {
    render(
        <BrowserRouter>
            <CreateBudgetPanel />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Submit/i);
    fireEvent.click(submit);
});
test('test cancel btn', () => {
    render(
        <BrowserRouter>
            <CreateBudgetPanel />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Cancel/i);
    fireEvent.click(submit);
});