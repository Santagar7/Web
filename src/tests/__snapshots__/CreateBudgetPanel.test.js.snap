// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="create_budget not_active"
>
  <div
    className="center"
  >
    <h1
      id="message"
    >
      Create new budget
    </h1>
    <form>
      <input
        data-testid="name"
        id="name"
        required={true}
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </form>
    <button>
      Cancel
    </button>
  </div>
</div>
`;
